src/proper_parenthetics/test_proper_parenthetics.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+"""Tests for Proper Parenthetics code challenge."""
+import pytest
+
+PARENTHETIC_TESTS = [
+    (("This is a () weird string( with )some parenthesis()"
+     "that is balanced()."), 0),
+    ("This is ))) a broken string )))(()).", -1),
+    ("This ((( string is (())(( open.", 1)
+]
```

Comment:
to really get at the issues you're having with your implementation, you'll need some more test cases.